### ISSUE TITLE
Re-run Table 4.7b (RT-IOT2022 open-set) on the de-leaked loader — full analysis, deferred

### ISSUE THREAD

**Issue by fundthmcalculus** [fundthmcalculus/grad-school#184]
**Deferred: schedule as a full analysis pass, not a quick re-quote.** Nothing here
blocks; Chapter 4's qualitative reading is expected to survive. This exists so the
re-run is not forgotten and so the analysis is done properly when it happens.

## Why

PR #183 removed a leaky column from `load_rt_iot2022`. The CSV's unnamed leading
column was being passed as an 82nd feature, and it is **not a row number**: the
file concatenates the twelve per-class captures and the counter restarts at zero
for each, so any value above 8,107 identifies `DOS_SYN_Hping` and nothing else.

Table 4.4 was unaffected — that row uses `mog_classifier`'s default `top_n = 5`
screen and the index column never ranked in the top five, so the same five
features are chosen either way (verified by re-running both ways).

**Table 4.7b is the row that consumed it.** It runs the antecedent screen over
*all* features and keeps all of them, so the leaky column went straight in. Worse,
PROVENANCE_MAP note 21 and `reproduce/outputs/OPENSET_COST_2026-08-22.md`
establish that this table's result is sensitive to the screen's **order** — and
removing a column perturbs the ranking, not just the feature set. So the effect is
not bounded by "one feature out of 82."

Its published numbers (+0.394 complement vs +0.537 Isolation Forest) therefore
still measure leaky data.

## What needs doing

- [ ] Re-run `table_4_4_openset.py` on RT-IOT2022 at the full seed count on the
      de-leaked loader (~82 min at the current pin, per note 21 — down from the
      3h38m the profiler's headline still quotes)
- [ ] Re-quote Table 4.7b wherever it appears: `04-fast-fis-synthesis-mog.md`
      (§4.3.5 and the Table 4.7b block), `appendix.md` A.7.2/A.7.3, `DATASETS.md`
- [ ] Re-generate the RT-IOT2022 θ-sweep / Fig 4.2 row if the headline moves
- [ ] PROVENANCE_MAP note recording the supersession

## Analysis to do at the same time, not separately

The reason this is marked "full analysis" rather than "re-run and re-quote":

1. **Seed-count confusion needs resolving.** Three different values are in play
   for the same table — the prose quotes **+0.394 / +0.537** from a 5-seed
   archive, `reproduce/outputs/table_4_4_openset.md` on disk holds a **3-seed**
   run reading **+0.333 / +0.513**, and note 21 quotes **+0.391 / +0.534** from
   the ten-seed θ-sweep. Whichever becomes the run of record, the others should be
   explicitly marked superseded rather than left to be found again.

2. **The D8 caveat is still open.** Note 21 says plainly: *"do not overwrite Table
   4.7b from this archive until D8 is settled"* — `_kmeans_labels_1d` swapped
   k-means++ for a uniform-random start upstream, and it reaches the complement
   rule. A re-run now measures current-pin code, not the code the prose quotes.
   This re-run and D8 should be settled together, or the same ambiguity is just
   re-created at a new set of digits.

3. **Order-sensitivity deserves a number.** Since the screen's order is known to
   matter and this change perturbs it, the re-run is a natural opportunity to
   report how much of any delta is the removed column versus the reordering —
   which note 21 flags as unconfirmed.

## Evidence in hand

A 1-seed probe on the de-leaked loader gives complement **+0.372**, one-class SVM
+0.437, Isolation Forest **+0.517**. A matched 1-seed control on the pre-fix
loader is being run to make that delta attributable; results will be added as a
comment. Early indication is that the gap to Isolation Forest persists at roughly
its published size, so the chapter's conclusion — the complement rule loses to
purpose-built detectors at this scale — is not expected to change. That is the
basis for deferring rather than blocking.

Follows #92 / #183.

**Comment by fundthmcalculus** [comment]
## Matched control run — the cost is 0.019 $J$, measured

Two single-seed `table_4_4_openset` runs, `REPRO_SEEDS=0`, same host, same day,
differing **only** in whether the loader drops the unnamed index column:

| Method | with index (82 feat.) | without (81 feat.) | Δ $J$ |
|---|---|---|---|
| **Complement rule** | +0.391 | **+0.372** | **−0.019** |
| One-class SVM | +0.444 | +0.437 | −0.007 |
| Isolation Forest | +0.517 | +0.517 | 0.000 |

**Why the control was necessary.** Neither of the values already lying around is
a valid baseline for a one-seed probe: the prose's **+0.394/+0.537** is five
seeds, and the `reproduce/outputs/table_4_4_openset.md` on disk is **three**
seeds at +0.333/+0.513. Comparing the probe to either would have charged seed
variance to the fix. Against a matched control, the delta is attributable.

**Reading.** The deficit to Isolation Forest **widens** rather than closes —
0.126 → 0.145, which brackets the 0.143 the prose already reports. Isolation
Forest is completely unmoved (identical to three decimals), which is expected:
it never depended on the antecedent screen. So the chapter's conclusion — the
complement rule loses to purpose-built detectors at this scale — is not at risk,
and this stays deferred.

**What this does not settle.** One seed is one seed. It bounds the effect; it
does not replace the five-seed table, and it says nothing about the three
analysis items above (seed-count reconciliation, D8, order-sensitivity
attribution). Those are still the reason this is a full-analysis pass.

Recorded in `PROVENANCE_MAP.md` note 25 addendum. PR #183 adds a reader-visible
caveat above Table 4.7b pointing here, so the PDF does not present those cells
as current in the meantime.

**Comment by fundthmcalculus** [comment]
## D8 is already settled by the pin we are on — the hold on re-quoting is lifted

Analysis item 2 above said this re-run had to wait for D8, quoting note 21's *"do
not overwrite Table 4.7b from this archive until D8 is settled."* That is stale.
**Both** causes have landed upstream and both are ancestors of the currently
pinned `353162c`:

| cause | fixed by | in the pin? |
|---|---|---|
| B14 — `wasserstein_distance` ignored the CDF gap width | `5253aa0` (#171) | yes |
| D8 — `_kmeans_labels_1d` single uniform-random start | `353162c` (#191) | **it *is* the pin** |

tribble-fis #191 ("swap safe candidates, keep hot-path ones", 2026-08-25) put
`sklearn.cluster.KMeans` back with `n_init="auto"`, restoring k-means++.

Verified in the running environment rather than from the diff — the stale-wheel
trap makes a diff insufficient evidence:

```
tribblefis file: C:\personal\grad-school\tribble-fis\src\tribblefis\__init__.py
uses sklearn KMeans: True
uses numba kmeans_1d: False
wasserstein weights by dx (#171): True
```

**So there is no upstream fix to write.** A ten-seed re-run at this pin is a
legitimate run of record for Table 4.7b, not a diagnostic to be reported beside
the prose.

### The pattern is worth more than the fix

This is the **third** instance of one failure mode. §7.3 already records it for
BETH — `TribbleOneClassDetector` arrived upstream and retired a stated blocker
while the proposal went on recording it as open. Note 22 drew the lesson (a pin
bump should ask which recorded blockers the new pin removes), and then that
lesson was not applied to D8, which the very next pin bump had settled.

`check_prose.py` catches *numbers* drifting between prose and harness. It cannot
see a **fix or capability** arriving in a submodule and silently retiring a
recorded blocker. That audit direction still has no automation behind it.

### Status

Ten-seed re-run on the de-leaked loader at this pin is running now (~90 min).
When it lands, Table 4.7b gets re-quoted and PR #183's caveat comes out — which
closes the substantive half of this issue. Remaining D8 scope is unrelated to
this table: the Chapter 3 compiler question (B15) and §4.3.2/G5 absorbing the
`pin_extremes` default flip (#102).

**Comment by fundthmcalculus** [comment]
## Ten-seed re-run done — Table 4.7b re-quoted, caveat removed

94 minutes, RT-IOT2022, leave-one-class-out, θ = 0.99, pin `353162c`, de-leaked
loader. Landed in #183 as `6bbbc00`.

| method | archive, 5 seeds | 2026-08-22, 10 seeds | **now, 10 seeds, de-leaked** |
|---|---:|---:|---:|
| Complement rule | +0.394 | +0.515 | **+0.366** |
| One-class SVM | +0.408 | +0.271 | **+0.410** |
| Isolation Forest | +0.537 | +0.579 | **+0.535** |

### The result that disappeared was the flattering one

CHECKLIST **C4** carried the 2026-08-22 column as *"the margin narrows sharply"* —
0.064 to Isolation Forest, with the complement rule **overtaking** the one-class
SVM. **Neither survives.** The margin is **0.169**, *wider* than the five-seed
archive's 0.143, and the complement rule trails the SVM again.

The leak and the stale pin were together flattering the construction, and
removing them costs it the one open-set result that read as a win. Worth saying
plainly given the direction.

### Attribution — honest and incomplete

Two things separate the 2026-08-22 column from this one: the leaky feature is
gone, **and** the pin restored k-means++ in `_kmeans_labels_1d` (#191). The
matched single-seed control puts the leak's own cost at **0.019 $J$**, so most of
the −0.149 is most plausibly the init restoration — but that is an *inference
from one seed against a ±0.27 spread, not a measurement*. Separating them needs a
matched ten-seed run at the old pin; not planned, and the document doesn't depend
on the split. This column is measured on correct data at a settled pin, which is
the only claim Table 4.7b makes.

### Unchanged

Spreads (±0.15–±0.27) still mean no separation clears its own error bar — exactly
what §4.4 already says of Table 4.7. §4.4's reading is *strengthened*: the free,
no-second-model property survives at scale, accuracy parity does not.

### Closing state

Done: re-run at the ten-seed floor, re-quote across Ch4/appendix/DATASETS.md,
PROVENANCE_MAP notes 26–27, CHECKLIST C4 superseded, PR #183 caveat removed.
Analysis items 1 and 2 are resolved — the seed-count confusion is reconciled in
the C4 table above, and D8 turned out to be already settled by the pin.

**Leaving this open for the remainder:** the RT-IOT2022 **θ-sweep** behind Fig
4.2's missing row was not regenerated (`REPRO_THETA_SWEEP` unset), so note 21's
sweep figures are still pre-de-leak and §4.4's "the sweep was not run here" still
stands. That plus analysis item 3 (order-sensitivity attribution) is what's left.

**Comment by fundthmcalculus** [comment]
## θ-sweep done — the gap is fundamental, not a threshold artefact (PR #202)

Ran the sweep on the de-leaked loader (2h09m, same invocation as the Table 4.7b
headline so they agree by construction). **Table 4.7c:**

| θ | Detection | False alarm | J |
|---:|---:|---:|---:|
| 0.80 | 0.914 | 0.518 | **+0.396 (peak)** |
| 0.99 | 0.804 | 0.438 | +0.366 (shipped) |
| Isolation Forest | 0.966 | 0.431 | **+0.535** |

**Answers the open question: tuning θ does not rescue the complement rule.**
Best-case +0.396 is a real but small +0.030 over shipped, and still **0.139 below
Isolation Forest**, which leads at every θ. The gap is fundamental to this
dataset. Peak sits at θ = 0.80 — exactly where the Glass sweep (Table 4.6) puts
its own, so the operating point transfers even though the verdict doesn't.

The leak-contaminated sweep found the same peak and non-closure (+0.391, gap
0.143), so the correction moved digits by thousandths. Recorded as PROVENANCE_MAP
note 28, superseding notes 21/27.

### Closing state

The three **analysis** items this issue was opened for are all resolved:
1. ✅ Seed-count reconciliation — done in the C4 table (5→10 seeds).
2. ✅ D8 — already settled by the pin (note 26).
3. ✅ Order-sensitivity — the de-leaked sweep matches the pre-leak one to
   thousandths, so the screen-order perturbation the leak fix caused is
   immaterial at the sweep level.

**Remaining, presentation-only:** whether to redraw Fig 4.2 with a second
(RT-IOT2022) panel. That's a figure choice, not a measurement gap — the numbers
are all in Table 4.7c. Happy to close this issue once that call is made, or leave
it open as the Fig 4.2 reminder.
